src/multi_processing/creator.py

```diff
@@ -0,0 +1,133 @@
+        of jobs, and puts a list of created records from that batch onto the
+        created record queue.
+    sleep_while_create_job_queue_empty()
+        Sleep whilst the create job queue is empty
+    get_dequeued_create_jobs(maximum_number_of_create_jobs_to_dequeue)
+        Return a list containing a batch of jobs dequeued from the create job
+        queue such that they can be run over a pool of child processes
+    """
+
+    def __init__(self, create_job_queue, created_record_queue):
+        """ Assign variables from input, and set termination to False
+
+        Parameters
+        ----------
+        create_job_queue : Multiprocessed Queue
+            Queue containing jobs to create a certain quantity of domain
+            objects
+        created_record_queue : Multiprocessed Queue
+            Queue containing resultant created records for writing to file
+        """
+
+        self.create_job_queue = create_job_queue
+        self.created_record_queue = created_record_queue
+        self.terminate_dequeued = False
+
+    def parent_process(self, object_factory):
+        """ Begin the cycle of waiting for, formatting, and running jobs,
+        continuing this until an instruction to terminate is observed.
+
+        Parameters
+        ----------
+        object_factory : Creatable
+            Instantiated and pre-configured object factory used to create
+            records from create jobs
+        """
+
+        number_of_create_child_processes =\
+            object_factory.get_shared_args()[
+                'number_of_create_child_processes'
+            ]
+
+        # the maximum number of jobs to dequeue for processing is proportional
+        # to the number of processes available to execute the jobs
+
+        maximum_number_of_create_jobs_to_dequeue = \
+            number_of_create_child_processes * 2
+
+        while not self.terminate_dequeued:
+            self.sleep_while_create_job_queue_empty()
+
+            dequeued_create_jobs = self.get_dequeued_create_jobs(
+                maximum_number_of_create_jobs_to_dequeue
+            )
+
+            created_records_from_multiple_jobs = \
+                pool_tasks.run_create_jobs(
+                    dequeued_create_jobs,
+                    number_of_create_child_processes,
+                    object_factory
+                )
+
+            self.created_record_queue.put(created_records_from_multiple_jobs)
+
+        self.created_record_queue.put("terminate")
+
+    def sleep_while_create_job_queue_empty(self):
+        """ Sleep until jobs are on the queue """
+
+        while self.create_job_queue.empty():
+            time.sleep(1)
+
+    def get_dequeued_create_jobs(
+            self, maximum_number_of_create_jobs_to_dequeue
+    ):
+        """ Takes jobs from the Multiprocess Safe Queue and places them into
+        a list for later execution. If twice the pool size of jobs are
```

Comment:
Explain why "If twice the pool size of jobs are formatted, then this list is returned"